bookshelf/3-binary-data/src/main/java/com/example/getstarted/util/CloudStorageHelper.java

```diff
@@ -54,22 +56,33 @@
    * Uploads a file to Google Cloud Storage to the bucket specified in the BUCKET_NAME
    * environment variable, appending a timestamp to end of the uploaded filename.
    */
-  @SuppressWarnings("deprecation")
+  // Note:This sample assumes small files are uploaded. For large files or streams use: 
+  // Storage.writer(BlobInfo blobInfo, Storage.BlobWriteOption... options)
   public String uploadFile(Part filePart, final String bucketName) throws IOException {
     DateTimeFormatter dtf = DateTimeFormat.forPattern("-YYYY-MM-dd-HHmmssSSS");
     DateTime dt = DateTime.now(DateTimeZone.UTC);
     String dtString = dt.toString(dtf);
     final String fileName = filePart.getSubmittedFileName() + dtString;
-
-    // the inputstream is closed by default, so we don't need to close it here
+    
+    // The InputStream is closed by default, so we don't need to close it here
+    // Read InputStream into a ByteArrayOutputStream.
+    InputStream is = filePart.getInputStream();
```

Comment:
I think this all can be done more simply: 
```java
FileInputStream file = new FileInputStream(fileName);
byte[] fileContent = Files.readAllBytes(file.toPath());
```

Then pass fileContent below.

Reply:
thanks! much simpler.

Reply:
Hey @kurtisvg, 

I jumped to quickly to add this code in.

The servlet handler doesn't provide the filePath and instead provides `filePart.getInputStream()` to access the uploaded data. I don't think this simplification will work in this case. 

Could you provide more context on this? I may have missed something but it not be usable in this case.

Reply:
I was looking at L65, which looked like it was getting the filename. I missed it was pulling from a Servlet stream. 

It looks like in Java 9, there is `InputStream.readAllBytes` but I'm guessing you still want to support Java 8. In that case, you'll probably have to do something like this: https://stackoverflow.com/a/1264737/9414803 (which you may already have been doing)

Reply:
Thanks @kurtisvg! +1 I saw Java 9 had a special helper, maybe in the future we can use it. For now, I'll revert and try to get Kokoro to pass.